RAGchain/benchmark/dataset/search_qa.py

```diff
@@ -0,0 +1,126 @@
+        Supporting metrics are 'Hole', 'TopK_Accuracy', 'EM', 'F1_score', 'Recall', 'Precision'
+        'context_recall', 'context_precision', 'BLEU', 'answer_relevancy', 'faithfulness', 'KF1'.
+        You must ingest all data for using context_recall and context_precision metrics.
+
+        Notice:
+        Context of SearchQA benchmark are all url in raw data. So we use lucadiliello's searchqa dataset at huggingface.
+        link: https://huggingface.co/datasets/lucadiliello/searchqa
+        Split taken from the MRQA 2019 Shared Task, formatted and filtered for Question Answering. For the original dataset,
+        have a look https://huggingface.co/datasets/mrqa.
+
+        Default metrics are essentially the metrics run when executing a test file.
+        Support metrics refer to the available metrics.
+        This distinction arises due to the prolonged evaluation time required for Ragas metrics.
+        """
+
+        self.file_path = "lucadiliello/searchqa"
+        self.dataset = load_dataset(self.file_path)['validation'].to_pandas()
+
+        default_metrics = self.retrieval_gt_metrics + self.answer_gt_metrics \
+                          + self.answer_no_gt_ragas_metrics + self.answer_passage_metrics
+        support_metrics = default_metrics + self.retrieval_gt_ragas_metrics + self.retrieval_no_gt_ragas_metrics
+
+        if metrics is not None:
+            # Check if your metrics are available in evaluation datasets.
+            for metric in metrics:
+                if metric not in support_metrics:
+                    raise ValueError(f"You input {metric} that this dataset evaluator not support.")
+            using_metrics = list(set(metrics))
+        else:
+            using_metrics = default_metrics
+
+        super().__init__(run_all=False, metrics=using_metrics)
+
+        self.eval_size = evaluate_size
+        self.run_pipeline = run_pipeline
+
+        # Delete duplicated question - answer - retrieval gt
+        self.dataset = self.dataset.groupby('question', as_index=False).agg(
+            {'context': lambda x: list(x), 'key': lambda x: list(x),
+             'answers': lambda x: list(x.tolist()), 'labels': lambda x: list(x)})
+
+        self.context = deepcopy(self.dataset[['key', 'context', 'labels']])
+
+        if evaluate_size is not None and len(self.dataset) > evaluate_size:
+            self.qa_data = self.dataset[:evaluate_size]
+
+    def ingest(self, retrievals: List[BaseRetrieval], db: BaseDB, ingest_size: Optional[int] = None):
+        """
+        Ingest dataset to retrievals and db.
+        :param retrievals: The retrievals that you want to ingest.
+        :param db: The db that you want to ingest.
+        :param ingest_size: The number of data to ingest. If None, ingest all data.
+        If you want to use context_recall and context_precision metrics, you should ingest all data.
```

Comment:
You don't need ingest all data for using context_precision.
context_preicision doesn't need retrieval gt contents.